fullscreen.bs

```diff
@@ -56,21 +57,22 @@ stated otherwise it is unset.
 
 <p>All <a for=/>documents</a> have an associated <dfn export>fullscreen element</dfn>. The
 <a>fullscreen element</a> is the topmost <a>element</a> in the <a for=/>document</a>'s
-<a>top layer</a> whose <a>fullscreen flag</a> is set, if any, and null otherwise.
+[=Document/top layer=] whose <a>fullscreen flag</a> is set, if any, and null otherwise.
 
 <p>All <a for=/>documents</a> have an associated <dfn>list of pending fullscreen events</dfn>, which
 is an <a>ordered set</a> of (<a>string</a>, <a>element</a>) <a>tuples</a>. It is initially empty.
 
-<p>To <dfn>fullscreen an <var>element</var></dfn>, set <var>element</var>'s <a>fullscreen flag</a>
-and <a for="top layer">add</a> it to its <a>node document</a>'s <a>top layer</a>.
+<p>To <dfn>fullscreen an <var>element</var></dfn>, set <var>element</var>'s <a>fullscreen flag</a>.
+If <var>element</var> is not currently [=in the top layer=] of its <a>node document</a>,
+[=add to the top layer|add it to the top layer=] of its <a>node document</a>.
 
 <p>To <dfn>unfullscreen an <var>element</var></dfn>, unset <var>element</var>'s
-<a>fullscreen flag</a> and <a>iframe fullscreen flag</a> (if any), and <a for=set>remove</a> it from
-its <a>node document</a>'s <a>top layer</a>.
+<a>fullscreen flag</a> and <a>iframe fullscreen flag</a> (if any), and
+<a>request removal from the top layer</a> for it from its <a>node document</a>.
```

Comment:
Yes, `:fullscreen` will immediately stop matching. I think this is good - it means if you have transitions set up for an exiting-fullscreen animation, they'll start going then. The difference between "Immediately" and "at next style update" aren't significant for this purpose; top layer needs to delay so it can react to newly-added 'overlay' transitions, but otherwise the CSS effects both happen at the same time, when you Update The Rendering.

Reply:
In practice, transitioning out of `:fullscreen` is quite difficult via CSS transitions anyway because we take out `position: fixed;`. It is also something the browser typically does a visual transition for too, which makes it hard to notice if any CSS transition is happening.

Personally, I would prefer going for "remove immediately from top layer" in this case, because exiting fullscreen is a critical operation (if someone tries to press "Esc", you don't want to allow top layer transitions) and it would be better perf wise too. Entering / exiting fullscreen is a fairly expensive operation style-wise in WebKit, especially when it involves subframes.

Reply:
Hm, that's a reasonable argument. I'm okay with removing it immediately when unfullscreening, as this is indeed somewhat of a special case.

@mfreed/@foolip does this sound okay to y'all too?

Reply:
Yes, I think we should do that. If we want to support CSS transitions when exiting fullscreen, we can follow up on that separately. I suspect additional changes would be needed.

Reply:
```suggestion
<a>remove from the top layer immediately</a> given |element|'s <a>node document</a> and |element|.
```